tt_metal/impl/program/program.cpp

```diff
@@ -107,7 +107,7 @@ void DisablePersistentKernelCache() { enable_persistent_kernel_cache = false; }
 std::atomic<uint64_t> Program::program_counter = 0;
 
 Program::Program() :
-    id(program_counter++), runtime_id(0), worker_crs_({}), local_circular_buffer_allocation_needed_(false), finalized_(false) {
+    id(program_counter++), runtime_id(0), worker_crs_({}), local_circular_buffer_allocation_needed_(false), finalized_(false), cached_(false) {
```

Comment:
Is the difference between `finalized` and `cached` that finalized means state should not change anymore and cached means the data structures for fast dispatch are populated on host?

Reply:
yes